streamwatcher_http.go

```diff
@@ -139,13 +145,12 @@ func (w *StreamWatcherHttp[StreamT]) watch(
 	close(outCh)
 }
 
-func (w *StreamWatcherHttp[StreamT]) Reconfigure(Buckets *StreamWatcherHttpConfig) error {
+func (w *StreamWatcherHttp[StreamT]) Reconfigure(cfg *StreamWatcherHttpReconfigureConfig) error {
 	w.lock.Lock()
 	w.state = &streamWatcherHttpState{
-		httpRoundTripper: Buckets.HttpRoundTripper,
-		endpoints:        Buckets.Endpoints,
-		userAgent:        Buckets.UserAgent,
```

Comment:
Why are we removing UserAgent from here?

Reply:
useragent shouldn't be a part of reconfiguration.